src/Knp/Menu/Twig/MenuExtension.php

```diff
@@ -36,6 +36,7 @@ public function getFilters(): array
     {
         return [
             new TwigFilter('knp_menu_as_string', [$this, 'pathAsString']),
+            new TwigFilter('knp_menu_spaceless', [$this, 'spaceless'], ['is_safe' => ['html']]),
```

Comment:
Shouldn't we rather deprecate our `compressed_root` block instead of providing our own spaceless filter suffering from the same shortcomings than the Twig one that has been deprecated because of them ?

Reply:
My initial idea was just about making out tests not failing about the twig deprecation, but I didn't manage to get it (using the environment variable SYMFONY_DEPRECATIONS_HELPER didn't work).
If we deprecated our internal block, we'll end up with the same problem (tests failing because of the deprecation)